lib/controllers/embed.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var jwt = require('jsonwebtoken');
+
+var winston = require.main.require('winston');
+var nbbAuthController = require.main.require('./src/controllers/authentication');
+var meta = require.main.require('./src/meta');
+var User = require.main.require('./src/user');
+
+
+var embedControllers = module.exports;
+
+
+embedControllers.embedView = function (req, res, next) {
+	meta.settings.get('openedx-discussion', function (err, settings) {
+		if (err) {
+			return next({
+				code: 'error',
+				plugin: 'openedx-discussion',
+				message: 'Settings could not be loaded',
+			});
+		}
+		var message = '';
+		if (!settings.hasOwnProperty('secret') || !settings.secret.length) {
+			message += '[nodebb-plugin-openedx-discussion] "secret" settings is not set.';
+			winston.error(message);
+		}
+		if (!settings.hasOwnProperty('jwtCookieName') || !settings.jwtCookieName.length) {
+			message = '[nodebb-plugin-openedx-discussion] "jwtCookieName" settings is not set.';
+			winston.error(message);
```

Comment:
I don't think we need to log the message here. Passing it forward as an error to `next()` should get the job done.

Reply:
Yes that makes sense but, nodebb usually logs its errors thats why I have  logged these messages too.